nmdc_runtime/site/translation/submission_portal_translator.py

```diff
@@ -444,4 +472,87 @@ def get_database(self) -> nmdc.Database:
 
+        if self.omics_processing_mapping:
+            # If there is data from an OmicsProcessing mapping file, process it now. This part
+            # assumes that there is a column in that file with the header __biosample_source_mat_id
+            # that can be used to join with the sample data from the submission portal. The
+            # biosample identified by that `source_mat_id` will be referenced in the `has_input`
+            # slot of the OmicsProcessing object. If a DataObject mapping file was also provided,
+            # those objects will also be generated and referenced in the `has_output` slot of the
+            # OmicsProcessing object. By keying off of the `source_mat_id` slot of the submission's
+            # sample data there is an implicit 1:1 relationship between Biosample objects and
+            # OmicsProcessing objects generated here.
+            join_key = "__biosample_source_mat_id"
+            database.omics_processing_set = []
+            database.data_object_set = []
+            data_objects_by_sample_data_id = {}
+            today = datetime.now().strftime("%Y-%m-%d")
+
+            if self.data_object_mapping:
+                # If DataObject mapping data was provided, group it by the sample ID key and then
+                # strip that key out of the resulting grouped data.
+                grouped = groupby(join_key, self.data_object_mapping)
+                data_objects_by_sample_data_id = valmap(
+                    lambda data_objects: [
+                        dissoc(data_object, join_key) for data_object in data_objects
+                    ],
+                    grouped,
+                )
+
+            for omics_processing_row in self.omics_processing_mapping:
+                # For each row in the OmicsProcessing mapping file, first grab the minted Biosample
+                # id that corresponds to the sample ID from the submission
+                sample_data_id = omics_processing_row.pop(join_key)
+                if (
+                    not sample_data_id
+                    or sample_data_id not in sample_data_to_nmdc_biosample_ids
+                ):
+                    logging.warning(
+                        f"Unrecognized biosample source_mat_id: {sample_data_id}"
+                    )
+                    continue
+                nmdc_biosample_id = sample_data_to_nmdc_biosample_ids[sample_data_id]
+
+                # Transform the raw row data according to the OmicsProcessing class's slots, and
+                # generate an instance. A few key slots do not come from the mapping file, but
+                # instead are defined here.
+                omics_processing_slots = {
+                    "id": self._id_minter("nmdc:OmicsProcessing", 1)[0],
+                    "has_input": [nmdc_biosample_id],
+                    "has_output": [],
+                    "part_of": nmdc_study_id,
+                    "add_date": today,
+                    "mod_date": today,
```

Comment:
I don't know whether the person who defined these slots expects the date to be in a particular timezone. I see, above, that the value is defined as `today = datetime.now().strftime("%Y-%m-%d")`. I think `datetime.now()` returns the local time with respect to the server (as opposed to, for example, the UTC time).

Reply:
That's a fair question. As far as I can tell from the existing data there is no particular standard. The schema doesn't [explicitly state](https://microbiomedata.github.io/nmdc-schema/add_date/) anything either. This might be something for further discussion later.

Reply:
OK, cool. Thanks for linking to the slot documentation—I don't see any info about timezones there either. I'll file a ticket in `nmdc-schema` as a starting point.